service/service_test.go

```diff
@@ -0,0 +1,253 @@
+			}
+		})
+	}
+}
+
+func TestService_deleteTeacher(t *testing.T) {
+	type fields struct {
+		rt *runtime.Runtime
+	}
+	type args struct {
+		id string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &Service{
+				rt: tt.fields.rt,
+			}
+			if err := s.deleteTeacher(tt.args.id); (err != nil) != tt.wantErr {
+				t.Errorf("deleteTeacher() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestService_retrieveStudent(t *testing.T) {
+	type fields struct {
+		rt *runtime.Runtime
+	}
+	type args struct {
+		id string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *models.Student
+		wantErr bool
+	}{}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &Service{
+				rt: tt.fields.rt,
+			}
+			got, err := s.retrieveStudent(tt.args.id)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("retrieveStudent() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("retrieveStudent() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestService_retrieveTeacher(t *testing.T) {
+	type fields struct {
+		rt *runtime.Runtime
+	}
+	type args struct {
+		id string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *models.Teacher
+		wantErr bool
+	}{}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &Service{
+				rt: tt.fields.rt,
+			}
+			got, err := s.retrieveTeacher(tt.args.id)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("retrieveTeacher() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("retrieveTeacher() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestService_updateStudent(t *testing.T) {
+	type fields struct {
+		rt *runtime.Runtime
+	}
+	type args struct {
+		student *models.Student
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &Service{
+				rt: tt.fields.rt,
+			}
+			if err := s.updateStudent(tt.args.student); (err != nil) != tt.wantErr {
+				t.Errorf("updateStudent() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestService_updateTeacher(t *testing.T) {
+	type fields struct {
+		rt *runtime.Runtime
+	}
+	type args struct {
+		teacher *models.Teacher
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &Service{
+				rt: tt.fields.rt,
+			}
+			if err := s.updateTeacher(tt.args.teacher); (err != nil) != tt.wantErr {
+				t.Errorf("updateTeacher() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
```

Comment:
Add todos on each test that is empty for the moment.

Reply:
But then linter says implement TODO, so I removed it for that reason.